### PR TITLE
fix(types): add `ExtendedModules` back to types

### PR DESCRIPTION
**Pull request by KorigamiK**

### CODE REVIEW

**Comment by lucario387** [comment]
Wait but why?
Shouldn't it be inside `Modules` directly now that ExtendedModules aren't a thing

**Comment by KorigamiK** [comment]
@lucario387 i think this fixs the issues in the generation

File: nvchad_types/all_hl_groups.lua [hunk 1]
```diff
@@ -501,7 +506,625 @@ error("Requring a meta file")
...
+---@field ["'@comment'"] Base46HLGroups # highlight group for treesitter
+---@field ["'@character'"] Base46HLGroups # highlight group for treesitter
+---@field ["'@attribute'"] Base46HLGroups # highlight group for treesitter
+---@field ["'@annotation'"] Base46HLGroups # highlight group for treesitter
 
 ---@class Base46HLGroupsList: HLGroups, ExtendedHLGroups
 
----@alias ExtendedModules
\ No newline at end of file
+---@alias ExtendedModules
```
Comment (lucario387) [inline nvchad_types/all_hl_groups.lua]: I believe `ExtendedModules` no longer sounds "correct", now that all are default and none are extended. Maybe renaming the alias (and its usage) to `ExtendedIntegrations` or `Base46Integrations` fits more

Comment (KorigamiK) [reply nvchad_types/all_hl_groups.lua]: I think `Base46Integrations` sounds good

**Review by lucario387 — APPROVED** [review]

**Comment by KorigamiK** [comment]
please merge as required @siduck

**Pull request merged**